components/mediation-ui/org.wso2.carbon.rest.api.ui/src/main/java/org/wso2/carbon/rest/api/ui/client/RestApiAdminClient.java

```diff
@@ -231,69 +236,67 @@ public void updateApiFromString(String apiName, String apiData) throws AxisFault
 		}
 	}
 
-     private String ReadWSDLPrefix(String svrURL){
-         try{
-			 String confPath = System.getProperty(CONF_LOCATION);
-			 if (confPath == null) {
-				 confPath = Paths.get("repository", "conf").toString();
-			 }
-			 InputStream in = new FileInputStream(Paths.get(confPath, "axis2", "axis2.xml").toString());
-			 OMElement results = OMXMLBuilderFactory.createOMBuilder(in).getDocumentElement();
-
-             AXIOMXPath xpathExpression = new AXIOMXPath ("/axisconfig/transportReceiver/parameter[@name='WSDLEPRPrefix']");
-             List nodeList = (List) xpathExpression.selectNodes(results);
-
-             if(!nodeList.isEmpty()){
-                 OMNode value = (OMNode) nodeList.get(0);
-                 String server = ((OMElementImpl) value).getText() ;
-
-                 if(server.contains("http") || server.contains("https")){
-                     return server;
-                 }
-             }
-         }catch(Exception e){
-             System.out.println(e.toString());
-         }
-
-         return "";
-     }
-
-     public String getServerContext() throws AxisFault {
-           try {
-                  String returnValue = ReadWSDLPrefix(stub.getServerContext());
-
-                  if(returnValue != ""){
-                           return returnValue;
-                  }else{
-                          return stub.getServerContext();
-                  }
-           } catch (Exception e) {
-                        handleException(bundle.getString("failed.to.get.servercontext"), e);
-           }
-         return null;
-     }
+	private String ReadWSDLPrefix() throws AxisFault {
+		try {
+			String confPath = System.getProperty(CONF_LOCATION);
+			if (confPath == null) {
+				confPath = Paths.get("repository", "conf").toString();
+			}
+			InputStream in = new FileInputStream(Paths.get(confPath, "axis2", "axis2.xml").toString());
+			OMElement results = OMXMLBuilderFactory.createOMBuilder(in).getDocumentElement();
+
+			AXIOMXPath xpathExpression =
+					new AXIOMXPath("/axisconfig/transportReceiver/parameter[@name='WSDLEPRPrefix']");
+			List nodeList = (List) xpathExpression.selectNodes(results);
+
+			if (!nodeList.isEmpty()) {

```

Comment:
Can nodeList be null ?

Reply:
Nope, in jaxen, it will always return an object as below.

return (List)(obj instanceof List ? (List)obj : new SingletonList(obj));